apps/web/src/app/(protectedPage)/forms/[formId]/_components/formEditor/formEditorCard.tsx

```diff
@@ -67,7 +68,13 @@ export function FormEditorCard({ form }: Readonly<Props>) {
   });
   const { isGeneratingAIField } = state;
 
-  const formDefaultValues = form as FormSubmitDataSchema;
+  const formDefaultValues = {
+    ...form,
+    formFields: form.formFields.map((field) => ({
+      ...field,
+      isMarkedForRemoval: false,
```

Comment:
Instead of mutating the original object, let's do it this way.

1. Create a state  `removeFieldsIds` to store all formIds here which need to be removed on form submit